aws/resource_aws_xray_sampling_rule.go

```diff
@@ -154,42 +159,63 @@ func resourceAwsXraySamplingRuleRead(d *schema.ResourceData, meta interface{}) e
 	d.Set("url_path", samplingRule.URLPath)
 	d.Set("version", samplingRule.Version)
 	d.Set("attributes", aws.StringValueMap(samplingRule.Attributes))
-	d.Set("arn", samplingRule.RuleARN)
+	d.Set("arn", arn)
+
+	tags, err := keyvaluetags.XrayListTags(conn, arn)
+	if err != nil {
+		return fmt.Errorf("error listing tags for Xray Sampling group (%q): %s", d.Id(), err)
+	}
+
+	if err := d.Set("tags", tags.IgnoreAws().IgnoreConfig(ignoreTagsConfig).Map()); err != nil {
+		return fmt.Errorf("error setting tags: %w", err)
+	}
 
 	return nil
 }
 
 func resourceAwsXraySamplingRuleUpdate(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).xrayconn
-	samplingRuleUpdate := &xray.SamplingRuleUpdate{
-		RuleName:      aws.String(d.Id()),
-		Priority:      aws.Int64(int64(d.Get("priority").(int))),
-		FixedRate:     aws.Float64(d.Get("fixed_rate").(float64)),
-		ReservoirSize: aws.Int64(int64(d.Get("reservoir_size").(int))),
-		ServiceName:   aws.String(d.Get("service_name").(string)),
-		ServiceType:   aws.String(d.Get("service_type").(string)),
-		Host:          aws.String(d.Get("host").(string)),
-		HTTPMethod:    aws.String(d.Get("http_method").(string)),
-		URLPath:       aws.String(d.Get("url_path").(string)),
+
+	if d.HasChange("tags") {
+		o, n := d.GetChange("tags")
+		if err := keyvaluetags.XrayUpdateTags(conn, d.Get("arn").(string), o, n); err != nil {
+			return fmt.Errorf("error updating tags: %w", err)
+		}
 	}
 
-	if d.HasChange("attributes") {
-		attributes := map[string]*string{}
-		if v, ok := d.GetOk("attributes"); ok {
-			if m, ok := v.(map[string]interface{}); ok {
-				attributes = stringMapToPointers(m)
+	if d.HasChanges("priority", "fixed_rate", "reservoir_size", "service_name", "service_type",
```

Comment:
This list needs to include `attributes` otherwise it'll never trigger below 👍  

```suggestion
	if d.HasChanges("attributes", "priority", "fixed_rate", "reservoir_size", "service_name", "service_type",
```

Reply:
Thanks!